enterprise/internal/a8n/service.go

```diff
@@ -747,7 +732,11 @@ func (s *Service) UpdateCampaign(ctx context.Context, args UpdateCampaignArgs) (
 		return campaign, nil, nil
 	}
 
-	if !campaign.Published() {
+	changesetCreation, err := tx.GetLatestChangesetJobCreatedAt(ctx, campaign.ID)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting latest changesetjob creation time")
+	}
+	if changesetCreation.IsZero() {
 		// If the campaign hasn't been published yet, we can simply update the
```

Comment:
Can you update this comment too? Something to this effect:

```suggestion
		// If not all ChangesetJobs have been created yet, that means the Campaign itself hasn't been published yet, but only individual changesets. In that case, we don't support update yet. See: https://github.com/sourcegraph/sourcegraph/issues/7915
```